File: languages/php/Bloock/Entity/Identity/CredentialProof.php
<?php

namespace Bloock\Entity\Identity;

/**
 * Represents the proof associated with a credential, including signature and sparse merkle tree proof.
 */
class CredentialProof
{
    private string $signatureProof;
    private string $sparseMtProof;

    /**
     * Constructs an CredentialProof object with the specified parameters.
     * @param string $signatureProof
     * @param string $sparseMtProof
     */
    public function __construct(string $signatureProof, string $sparseMtProof)
    {
        $this->signatureProof = $signatureProof;
        $this->sparseMtProof = $sparseMtProof;
    }

    public static function fromProto(\Bloock\CredentialProof $res): CredentialProof
    {
        return new CredentialProof($res->getSignatureProof(), $res->getSparseMtProof());
    }

    /**
     * Retrieve signature proof with string format
     * @return string
     */
    public function getSignatureProof(): string
    {
        return $this->signatureProof;
    }

    /**
     * Retrieve sparse merkle tree proof with string format
     * @return string
     */
    public function getSparseMtProof(): string
    {
        return $this->sparseMtProof;
    }

    public function toProto(): \Bloock\CredentialProof
    {
        $p = new \Bloock\CredentialProof();

        if ($this->signatureProof != null) {
            $p->setSignatureProof($this->signatureProof);
        }

        if ($this->sparseMtProof != null) {
            $p->setSparseMtProof($this->sparseMtProof);
        }
    
        return $p;
    }
}

File: languages/python/bloock/entity/identity/credential_proof.py
from __future__ import annotations

import bloock._bridge.proto.identity_entities_pb2 as proto


class CredentialProof:
    """
    Represents the proof associated with a credential, including signature and sparse merkle tree proof.
    """
    def __init__(self, signature_proof: str, sparse_mt_proof: str | None) -> None:
        """
        Constructs an CredentialProof object with the specified parameters.
        :type sparse_mt_proof: object
        :type signature_proof: object
        :rtype: object
        """
        self.signature_proof = signature_proof
        self.sparse_mt_proof = sparse_mt_proof

    @staticmethod
    def from_proto(c: proto.CredentialProof) -> CredentialProof:
        sparse_mt_proof = None
        if c.sparse_mt_proof is not None:
            if c.sparse_mt_proof != '':
                sparse_mt_proof = c.sparse_mt_proof
        return CredentialProof(
            signature_proof=c.signature_proof,
            sparse_mt_proof=sparse_mt_proof,
        )

    def to_proto(self) -> proto.CredentialProof:
        sparse_mt_proof = None
        if self.sparse_mt_proof is not None or self.sparse_mt_proof != '':
            sparse_mt_proof = self.sparse_mt_proof

        return proto.CredentialProof(
            signature_proof=self.signature_proof,
            sparse_mt_proof=sparse_mt_proof,
        )
